new_platforms/include/optee/cyres_optee.h

```diff
@@ -0,0 +1,16 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* Licensed under the MIT License. */
+#pragma once
+
+oe_result_t get_cyres_seal_secret(
+    const uint8_t* key_selector,
+    size_t key_selector_size,
+    uint8_t** secret,
+    size_t* secret_size,
+    size_t req_size);
+
+oe_result_t get_cyres_private_key(uint8_t** key, size_t* key_size);
+
+oe_result_t get_cyres_public_key(uint8_t** pem, size_t* pem_size);
+
+oe_result_t get_cyres_cert_chain(uint8_t** pem, size_t* pem_size);
```

Comment:
Result is a null-terminated string, so consider changing to char ** and removing size parameter. Buffer is callee allocated so caller shouldn't need to care about size.

Reply:
I would keep the API as is for flexibility. The API that I have for `oe_get_*_key` has a parameter that specifies the output type. Right now, it just has PEM as the output but later on we might add more. For example, I imagine a common use case would be signing with the private key, so we might return a public API version of the `oe_private_key_t` struct.